src/linker/Linker/Driver.cs

```diff
@@ -500,6 +500,41 @@ protected int SetupContext (ILogger customLogger = null)
 
 						continue;
 
+					case "--collapse":
+					case "--collapse+": {
+							string assemblyName = GetNextStringValue ();
+							if (!string.IsNullOrEmpty (assemblyName)) {
+								if (!IsValidAssemblyName (assemblyName)) {
```

Comment:
Maybe I'm missing something, but I don't think this will ever evaluate true. Unrelated to your changes, but I'm also a little bit confused why we need this `IsValidAssemblyName` method? It doesn't seem to be evaluating the validity of anything.

Reply:
Thanks. I think the intention of IsValidAssemblyName is to catch empty strings passed on the command-line, so I updated the outer check to check null only. This way if you pass `--collapse ""` you get a warning about the assembly name.